qiskit/pulse/calibration_entries.py

```diff
@@ -64,6 +64,15 @@ def get_schedule(self, *args, **kwargs) -> Union[Schedule, ScheduleBlock]:
         """
         pass
 
+    @abstractmethod
```

Comment:
Can you do this without breaking compatibility? Wouldn't adding a new required abstract method break any downstream usage of `CalibrationEntry`?

Reply:
Done in [33b76b3](https://github.com/Qiskit/qiskit-terra/pull/9587/commits/33b76b3598d5bb57454383f820c4dd8243fb8249)

Reply:
Previously I implemented an abstractmethod to return schedule duration. Note that schedule duration is computed here for added schedule from the inst map, but this requires Qobj parsing that is the biggest performance bottleneck for non-pulse users. So I have implemented a custom (complicated) method that compute duration without parsing Qobj.
https://github.com/Qiskit/qiskit-terra/blob/bf3bb96cebc141f79ce46b5938724eef30778c8e/qiskit/transpiler/target.py#L449-L457

New abstract method just returns whether it is provided by backend. If your schedule is provided from backend, you should know duration of it and we don't need to update duration in the first place. This avoids unnecessary Qobj parsing for duration.